### ISSUE TITLE
Unreachable imports not ignored in methods

### ISSUE THREAD

**Issue by JukkaL** [python/mypy#4324]
The second `import xxx as os` generates an error, though it's similar to the first import, which doesn't generate an error:

```py
import sys

def f() -> None:
    if sys.platform == 'xyz':  # Nonsensical platform
        import xxx as os  # No error
    else:
        import os

class A:
    def g(self) -> None:
        if sys.platform == 'xyz':
            import xxx as os  # Error
        else:
            import os
```

The culprit is semantic analyzer pass 1, which only recurses into top-level functions but not into methods.

**Comment by JukkaL** [comment]
This has been fixed in the new semantic analyzer.
